vulture/core.py

```diff
@@ -96,6 +96,20 @@ def _ignore_variable(filename, varname):
         _is_special_name(varname))
 
 
+def condition_is_unsatisfiable(condition):
+    """
+    Checks if the given condition is unsatisfiable 
+    """
+    try:
+        cond = condition if isinstance(condition, ast.Expression) else ast.Expression(condition)
+        test = compile(cond, filename="<string>", mode='eval')
+        satisfiable = bool(eval(test, {}, {}))
+    except (NameError, ValueError):
```

Comment:
We also need to catch TypeError. We need test for all errors, please add tests with comments about which errors they trigger:

    'chr(-1)'   # raises ValueError
    'getattr(True, "foo")'  # raises AttributeError
    'isinstance(True, True)  # raises TypeError

'foo' and 'foo and False' raise NameError.